run.py

```diff
@@ -50,56 +51,67 @@ def run_one_step_with_cudastreams(func, streamcount):
 
         print(f"Cuda StreamCount:{len(streamlist)}: gpu time {start_event.elapsed_time(end_event)}")
 
+
 def run_one_step(func):
 
     func()
 
     if args.device == "cuda":
-      torch.cuda.synchronize()
-      start_event = torch.cuda.Event(enable_timing=True)
-      end_event = torch.cuda.Event(enable_timing=True)
+        torch.cuda.synchronize()
+        start_event = torch.cuda.Event(enable_timing=True)
+        end_event = torch.cuda.Event(enable_timing=True)
 
-      start_event.record()
+        start_event.record()
 
-      t0 = time.time()
-      func()
-      t1 = time.time()
+        t0 = time.time()
+        func()
+        t1 = time.time()
 
-      end_event.record()
-      torch.cuda.synchronize()
-      print(f"Ran in {t1 - t0} seconds, gpu time {start_event.elapsed_time(end_event)}.")
+        end_event.record()
+        torch.cuda.synchronize()
+        print(f"Ran in {t1 - t0} seconds, gpu time {start_event.elapsed_time(end_event)}.")
 
     else:
 
-      t0 = time.time()
-      func()
-      t1 = time.time()
+        t0 = time.time()
+        func()
+        t1 = time.time()
 
-      print(f"Ran in {t1 - t0} seconds.")
+        print(f"Ran in {t1 - t0} seconds.")
 
 
 def profile_one_step(func, nwarmup=3):
-    for i in range(nwarmup):
-        func()
-
-    use_cuda = args.device == "cuda"
-
-    with profiler.profile(record_shapes=True, use_cuda = use_cuda) as prof:
-        func()
+    activity_groups = []
+    if args.device == "cuda":
+        activity_groups.append(profiler.ProfilerActivity.CUDA)
+    activity_groups.append(profiler.ProfilerActivity.CPU)
+
+    with profiler.profile(
+        schedule=profiler.schedule(wait=0, warmup=nwarmup, active=1),
+        activities=activity_groups,
+        record_shapes=True,
+        on_trace_ready=profiler.tensorboard_trace_handler(args.profile_folder)
+    ) as prof:
+        for _i in range(nwarmup + 1):
+            func()
+            prof.step()
```

Comment:
Just curious that you are now running `prof.step()` with warmup loops, is there a reason for that?

Reply:
I think both ways work the same, I need to look into if profiler does anything special for warmup steps. I think this is the expected way to use `torch.profiler` from the docs. This change reflects the docs better, but is not necessary. Do you have a preference? I don't mind changing it back.

Reply:
Either way looks okay to me too :)